Lib/pickle.py

```diff
@@ -1814,4 +1814,5 @@ def _test():
             import pprint
             for f in args.pickle_file:
                 obj = load(f)
+                f.close()
```

Comment:
This will not be called if `load` causes exception for cases like invalid payload. Similar case with the approach in `pickletools` too. Sometimes even if one file causes exception the other files in the loop will not be iterated and closed.

```
$ echo invalid > mypickle.pickle
$ ./python -Wall -m pickle mypickle.pickle
Traceback (most recent call last):
  File "/root/cpython/Lib/runpy.py", line 197, in _run_module_as_main
    return _run_code(code, main_globals, None,
  File "/root/cpython/Lib/runpy.py", line 87, in _run_code
    exec(code, run_globals)
  File "/root/cpython/Lib/pickle.py", line 1816, in <module>
    obj = load(f)
_pickle.UnpicklingError: could not find MARK
sys:1: ResourceWarning: unclosed file <_io.BufferedReader name='mypickle.pickle'>
```

Reply:
hmm.. good catch. I'll use another approach.